tools/xdmod-ondemand-export/xdmod_ondemand_export/__init__.py

```diff
@@ -389,6 +395,19 @@ def __entry_time_field_to_str(self, time_fields, key):
             else '-'
         )
 
+    def __get_ip_hash(self, ip):
```

Comment:
My concern is that this file could potentially grow very large - and also the performance of the dict lookup could be pretty poor when there are lots of IP addresses in the logs.

How about using a HMAC hash (with the secret generated and stored at the remote site but not shared with us)?  https://ee.lbl.gov/LBNL-FTP-ANON.html 

something like this

```
#!/usr/bin/env python3

import hmac
import secrets
import base64

# real code would generate this and store in a file for reuse
key = secrets.token_bytes(16)

# example ip address string from logs
ipaddr = "127.0.0.1"

# b85 encode to avoid non-printable characters while trying to keep the
# string size shortish
anon_hash = base64.a85encode(hmac.digest(key,  bytes(ipaddr, 'utf-8'), 'md5'))

print(anon_hash)
```

Reply:
Thank you, I have switched it to use an HMAC hash, but with Base64 encoding so the log parser in the ingestor can parse the value.